src/arrayMethodShift.js

```diff
@@ -6,8 +6,11 @@
  */
 function applyCustomShift() {
   [].__proto__.shift2 = function() {
-    // write code here
+    if (this.length === 0) {
+      return undefined;
+    };
+    return this.splice(0, 1)[0];
```

Comment:
The goal of this task is to understand how array methods work inside. I'm not sure you understand that when you use some other method.